saa/1.1. Interacting with services - AWS CLI & SDK.md

```diff
@@ -34,19 +34,20 @@
       - If EC2 is shared -> other people may perform actions impersonating you
   - 💡 Better way: AWS IAM Roles
     - IAM Roles can be attached to EC2 instances.
-      - ❗ One EC2 instance can have one IAM role at a time
+      - ❗ One EC2 instance can have one IAM role at a time(but the same role can be attached to many other EC2 instances)
+      - 📝 Also, Roles are not directly attached to EC2 instance, but are attached to instance profile. Instance profile is a container, where roles are attached.
     - IAM Roles can come with a policy authorizing exactly what the EC2 instance should be able to do e.g. AdminAccess or List buckets in S3.
     - EC2 instances can use these profiles automatically without any additional configurations
   - 📝If EC2 need access to other aws resources, never put credentials in it, always use IAM roles
   - Flow:
     - Create IAM role for the resource EC2 will access
     - EC2 portal -> Right click on instance -> Settings -> Attach IAM role
 - **S3 CLI**
-  - `aws s3 ls`: Lists all buckets available
-  - `aws s3 ls s3://bucketname`: Lists contents of the bucket
-  - `aws s3 cp s3://bucketname/pic.jpg pic.jpg`: Copy a local file, or S3 objects to your computer or from your computer.
-  - `aws s3 mb s3://newbucket`: create a new bucket
-  - `aws s3 rb s3://newbucket`: delete existing bucket
+  - `aws s3 ls --region us-east-1`: Lists all buckets available
+  - `aws s3 ls s3://bucketname --region us-east-1`: Lists contents of the bucket
+  - `aws s3 cp s3://bucketname/pic.jpg --region us-east-1`: Copy a local file, or S3 objects to your computer or from your computer.
```

Comment:
`cp` requires two file arguments maybe add back `pic.jpg` such as: `aws s3 cp pic.jpg s3://bucketname/pic.jpg`

Reply:
When I made in a hurry, I missed it, glad you are here to note that.